packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -170,22 +208,75 @@ async function updateWalletAddress(
   }
 }
 
+async function getWalletAddressWithAdditionalProperties(
+  deps: ServiceDependencies,
+  id: string,
+  includeVisibleOnlyAddProps: boolean
+): Promise<WalletAddress | undefined> {
+  return getWalletAddressBy(deps, id, true, includeVisibleOnlyAddProps)
+}
+
 async function getWalletAddress(
   deps: ServiceDependencies,
   id: string
 ): Promise<WalletAddress | undefined> {
-  return await WalletAddress.query(deps.knex)
+  return getWalletAddressBy(deps, id, false, false)
+}
+
+async function getWalletAddressBy(
+  deps: ServiceDependencies,
+  id: string,
+  includeAddProps: boolean,
+  includeVisibleOnlyAddProps: boolean
+): Promise<WalletAddress | undefined> {
+  const returnVal = await WalletAddress.query(deps.knex)
     .findById(id)
     .withGraphFetched('asset')
+
+  if (includeAddProps && returnVal) {
+    returnVal.additionalProperties = await getWalletAdditionalProperties(
+      deps,
+      returnVal.id,
+      includeVisibleOnlyAddProps
+    )
+  }
+  return returnVal
+}
```

Comment:
Not blocking and maybe just my opinion, but I think it would be cleaner to just have separate `getWalletAdditionalProperties` and `getWalletAddress` methods without this central one.